packages/opentelemetry-cloud-trace-propagator/test/CloudPropagator.test.ts

```diff
@@ -0,0 +1,210 @@
+  });
+
+  describe('.extract()', () => {
+    it('should extract context of a sampled span from carrier', () => {
+      carrier[X_CLOUD_TRACE_HEADER] =
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=1';
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.SAMPLED,
+        isRemote: true,
+      });
+    });
+
+    it('should extract context of a unsampled span from carrier', () => {
+      carrier[X_CLOUD_TRACE_HEADER] =
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=0';
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.NONE,
+        isRemote: true,
+      });
+    });
+
+    it('should extract specified traceId and auto-generated spanId.', () => {
+      carrier[X_CLOUD_TRACE_HEADER] = 'b75dc0042a82efcb6b0a194911272926';
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+
+      assert.deepStrictEqual(
+        extractedSpanContext!.traceId,
+        'b75dc0042a82efcb6b0a194911272926'
+      );
+      assert.deepStrictEqual(extractedSpanContext!.traceFlags, TraceFlags.NONE);
+    });
+
+    it('should handle missing trace_flags', () => {
+      carrier[X_CLOUD_TRACE_HEADER] =
+        'b75dc0042a82efcb6b0a194911272926/1258215';
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+
+      assert.deepStrictEqual(
+        extractedSpanContext!.traceId,
+        'b75dc0042a82efcb6b0a194911272926'
+      );
+      assert.deepStrictEqual(extractedSpanContext!.spanId, '00000000001332e7');
+      assert.deepStrictEqual(extractedSpanContext!.traceFlags, TraceFlags.NONE);
+    });
+
+    it('should extract x-cloud-trace-context from list of header', () => {
+      carrier[X_CLOUD_TRACE_HEADER] = [
+        'd4cda95b652f4a1592b449d5929fda1b/7929822056569588882;o=1',
+      ];
+      const extractedSpanContext = getExtractedSpanContext(
+        cloudPropagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
+      );
+      assert.deepStrictEqual(extractedSpanContext, {
+        traceId: 'd4cda95b652f4a1592b449d5929fda1b',
+        spanId: '6e0c63257de34c92',
+        traceFlags: TraceFlags.SAMPLED,
+        isRemote: true,
+      });
+    });
+
+    it('should auto-generated IDs when X-Cloud-Trace-Context: o=1', () => {
+      carrier[X_CLOUD_TRACE_HEADER] = 'o=1';
```

Comment:
This test should fail, as the semicolon is required.

Reply:
**X-Cloud-Trace-Context: o=1 trace and use auto-generated IDs.** This is as per https://www.npmjs.com/package/gcloud-trace. Also, suggested by @sk-

Reply:
That NPM package is not an authoritative reference; it's developed by an individual 3rd party developer. The info I shared comes from internal Google unit tests.